langfuse/utils/compile_template_string.py

```diff
@@ -0,0 +1,37 @@
+def compile_template_string(content: str, **kwargs) -> str:
+    opening = "{{"
+    closing = "}}"
+
+    result_list = []
+    curr_idx = 0
+
+    while curr_idx < len(content):
+        # Find the next opening tag
+        var_start = content.find(opening, curr_idx)
+
+        if var_start == -1:
+            result_list.append(content[curr_idx:])
+            break
+
+        # Find the next closing tag
+        var_end = content.find(closing, var_start)
+
+        if var_end == -1:
+            result_list.append(content[curr_idx:])
```

Comment:
This here would happen in case of a male formatted sting, right? `{{var1}} something{{random {{var2}}` would not match var2, right?